src/Core/FakeVimCommand.cpp

```diff
@@ -0,0 +1,288 @@
+                 appwin->currentWindow()->testcases->count() >= out)
+        {
+            appwin->currentWindow()->runTestCase(out - 1);
+        }
+        else // this branch is actually else-if, if is coming from macro
+            SHOW_STATUS_MESSAGE(tr("Error: cannot run testcase %1").arg(cmd.args));
+        return true;
+    }
+    if (wantKillAll(cmd))
+    {
+        appwin->on_actionKillProcesses_triggered();
+        return true;
+    }
+    if (wantSnippets(cmd))
+    {
+        appwin->on_actionUseSnippets_triggered();
+        return true;
+    }
+    if (wantFormat(cmd))
+    {
+        appwin->on_actionFormatCode_triggered();
+        return true;
+    }
+    if (wantVModeChange(cmd).first)
+    {
+        auto val = wantVModeChange(cmd).second;
+        if (val == "split")
+            appwin->on_actionSplitMode_triggered();
+        if (val == "edit")
+            appwin->on_actionEditorMode_triggered();
+        return true;
+    }
+    if (wantPreferences(cmd))
+    {
+        appwin->on_actionSettings_triggered();
+        return true;
+    }
+    if (wantLanguageChange(cmd).first)
+    {
+        auto val = wantLanguageChange(cmd).second;
+        if (appwin->currentWindow() && !val.isEmpty())
+            appwin->currentWindow()->setLanguage(val);
+        return true;
+    }
+    if (wantMessageLoggerClear(cmd) && appwin->currentWindow())
+    {
+        appwin->currentWindow()->on_clearMessagesButton_clicked();
+        return true;
+    }
+    if (wantExit(cmd))
+    {
+        appwin->on_actionQuit_triggered();
+        return true;
+    }
+    // return false indicating that custom command handler failed to handle the command
+    return false;
+}
+
+QPair<bool, QString> FakeVimCommand::wantNewFile(FakeVim::Internal::ExCommand const &ex)
+{
+    if (ex.cmd == "new")
+    {
+        if (ex.args == "c++" || ex.args == "cpp")
+            return {true, "C++"};
+        else if (ex.args == "py" || ex.args == "python")
+            return {true, "Python"};
+        else if (ex.args == "java")
+            return {true, "Java"};
+        else if (ex.args.isEmpty())
+            return {true, SettingsHelper::getDefaultLanguage()};
+        else
+            SHOW_STATUS_MESSAGE(
+                tr("Error: %1 requires an empty or one of \"cpp\", \"java\" and \"python\" argument").arg(ex.cmd));
+    }
+
+    return {false, ""};
+}
+
+QPair<bool, QString> FakeVimCommand::wantOpenFile(FakeVim::Internal::ExCommand const &ex)
+{
+    if (ex.cmd == "open" || ex.cmd == "opn")
+    {
+        if (ex.args.isEmpty())
+        {
+            SHOW_STATUS_MESSAGE(tr("Error: %1 requires a file path").arg(ex.cmd));
+            return {true, ""};
+        }
+
+        QFileInfo file(ex.args);
+        if (!file.exists())
+        {
+            SHOW_STATUS_MESSAGE(tr("Error: %1 does not exists").arg(ex.args));
+            return {true, ""};
+        }
```

Comment:
> Actually CP Editor can open a tab with a non-exist file path.

`open` and `new` are two different commands. One is used to edit existing files and hence requires file path to exist. `new` is like opening new buffer. Doesn't need to have file path.

How do you open a file that doesn't exists with `Ctrl+O`?

Reply:
This behavior is usually done by running `cpeditor <a non-exist path>` in the command line. It's a quick way to set the file name when creating a new tab. It's not available via Ctrl+O because QFileDialog only allows existing files.

Reply:
This is documented [here](https://cpeditor.org/docs/tips/#launch-cp-editor-in-the-command-line).

Reply:
`open` command emulates `Ctrl+O` also vim equivalent for open is `:e` which opens only existing file paths. 

I never tried  `cpeditor <path>` but it seems like it creates a path and opens in new tabs. (Open should not create files)

Reply:
1. Since `open` emulates Ctrl+O, I think it can open a file dialog when arguments are empty.
2. No file is created when opening a non-exist file. It's simply a quick way to set the path used when saving it. It's the same as `vim <non-exist path>`.

Reply:
>  No file is created when opening a non-exist file. It's simply a quick way to set the path used when saving it. It's the same as `vim <non-exist path>`

This is a good approach for non existing paths. So how exactly do we set such path in new tabs?

Reply:
> So how exactly do we set such path in new tabs?

Nothing different from an existing path.

Reply:
I think default behaviour of open should be to error out in non-existing paths. However, I can add handling `!` (bang) command so that it forcefully opens a path. An example would be `open! <non-existing-path>`